Formula/p/polkit.rb

```diff
@@ -15,15 +21,32 @@ class Polkit < Formula
   depends_on "ninja" => :build
   depends_on "pkg-config" => [:build, :test]
   depends_on "duktape"
-  depends_on "expat"
   depends_on "glib"
-  depends_on :linux # macOS requires some patches to build due to installing into /usr/lib
-  depends_on "linux-pam"
-  depends_on "systemd"
+  uses_from_macos "expat"
+
+  on_macos do
+    depends_on "gettext"
+  end
+
+  on_linux do
+    depends_on "linux-pam"
+    depends_on "systemd"
+  end
 
   def install
-    args = ["-Dsystemdsystemunitdir=#{lib}/systemd/system"]
+    inreplace "meson.build" do |s|
+      s.gsub!("sysusers_dir = '/usr/lib/sysusers.d'", "sysusers_dir = '#{etc}/sysusers.d'")
+      s.gsub!("tmpfiles_dir = '/usr/lib/tmpfiles.d'", "tmpfiles_dir = '#{etc}/tmpfiles.d'")
+    end
+
+    args = [
+      "-Dsystemdsystemunitdir=#{lib}/systemd/system",
+      "-Dpam_prefix=#{etc}/pam.d",
+      "-Dpam_module_dir=#{lib}/pam",
+    ]
+    args << "-Dsession_tracking=ConsoleKit" if OS.mac?
 
+    ENV.append_to_cflags "-Wno-implicit-function-declaration" if DevelopmentTools.clang_build_version >= 1403
```

Comment:
This should probably have been `-D_DARWIN_C_SOURCE` as it needs functions guarded by that.

Related: https://github.com/macports/macports-ports/blob/master/security/policykit/files/patch-src__polkitbackend__meson.build-define-darwin-c-source.diff

From `netdb.h`:
```h
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
void		freehostent(struct hostent *);
struct hostent	*gethostbyname2(const char *, int);
struct hostent	*getipnodebyaddr(const void *, size_t, int, int *);
struct hostent	*getipnodebyname(const char *, int, int, int *);
struct rpcent	*getrpcbyname(const char *name);
#ifdef __LP64__
struct rpcent	*getrpcbynumber(int number);
#else
struct rpcent	*getrpcbynumber(long number);
#endif
struct rpcent	*getrpcent(void);
void		setrpcent(int stayopen);
void		endrpcent(void);
void		herror(const char *);
const char	*hstrerror(int);
int			innetgr(const char *, const char *, const char *, const char *);
int			getnetgrent(char **, char **, char **);
void		endnetgrent(void);
void		setnetgrent(const char *);
#endif /* (!_POSIX_C_SOURCE || _DARWIN_C_SOURCE) */
```